qutebrowser/utils/utils.py

```diff
@@ -864,6 +864,63 @@ def unused(_arg):
     pass
 
 
+def parse_numsettxt_into_list(txtset, nummax):
```

Comment:
The asymptotic worst run time complexity of this function is O(max(txtset)). E.g. if someone defines txtset = 1,3,5-100000, then max(txtset) = 100000.

Once this potentially long list is generated, other commands will evaluate them one by one (that's how I did them so far). E.g. `download-delete 1,3,5-100000` will cause a loop to be run in `_select_downloads` that will have 100009 iterations. So because of this, the worst run time complexity of `_select_downloads` is also O(max(txtset)).

---

Alternatively, I can change this, so that the asymptotic worst run time becomes O(number of downloads). So if we have, say 50 download objects above the tabs, then we will have 50 iterations.

If I rub more neurons, I can probably get it down to O(min(txtset, number of downloads)).

But the method that is in my mind, while will allow us to change the complexity from O(max(txtset)) down to O(number of downloads), or even O(min(txtset, number of downloads)), would probably be more expensive per each step.

Basically, `parse_numsettxt_into_list` will become `is_number_in_numsettxt(num, txtset)`, and it will return `True` or `False` if number `num` is in the set described by the text `txtset`. This way, if the user defines some crazy set like `1-112342342534`, we will not return a numbers list that has all those crazy numbers.

Or maybe to reduce function calls, `is_numbers_in_numsettxt(numbers, txtset)`, where the function will return two sets of numbers, one that contains numbers that exist in `txtset`, and another that contains those that don't.

----

Or maybe I am missing a much simpler solution.